lib/lib_install.php

```diff
@@ -75,6 +75,103 @@ function checkRequirements(string $dbType = ''): array {
 	);
 }
 
+function getProcessUsername(): string {
+	if (function_exists('posix_getpwuid') && function_exists('posix_geteuid')) {
+		$processUser = posix_getpwuid(posix_geteuid()) ?: [];
+		if (!empty($processUser['name'])) {
+			return $processUser['name'];
+		}
+	}
+
+	if (function_exists('exec')) {
+		exec('whoami', $output);
+		if (!empty($output[0])) {
+			return $output[0];
+		}
+	}
+
+	return _t('install.check.unknown_process_username');
+}
+
+/** @return array<string> */
+function getOwnerOfFile(String $file): array {
+	clearstatcache();
+	$stat_file = @stat($file);
+	$owner = [];
+	if(!$stat_file) {
+		//Couldnt stat file
+		$owner['fileowner'] = _t('install.check.unknown_process_username');
+		$owner['filegroup'] = _t('install.check.unknown_process_username');
+		return $owner;
+	}
+	if (function_exists('posix_getpwuid') && function_exists('posix_getgrgid')) {
+		$uid_name = @posix_getpwuid($stat_file['uid']);
+		$gid_name = @posix_getgrgid($stat_file['gid']);
+		$owner['fileowner'] = is_array($uid_name) ? $uid_name['name'] : _t('install.check.unknown_process_username');
+		$owner['filegroup'] = is_array($gid_name) ? $gid_name['name'] : _t('install.check.unknown_process_username');
+		return $owner;
+	};
```

Comment:
@Alkarex 
Line 107: Is it not enough?

Reply:
It probably is - I have not read the PR yet. We just need to be extra careful for things that are not natively cross-platform, to test them in multiple configurations